### PR TITLE
feat: #1081 grant application admin component test

### PR DESCRIPTION
**Pull request by J0taFerreira**
- tested the breadcrumb, page title, sub title, submit button is disabled initially
- tested the validation error and message for user id field
- tested breadcrumb item redirect on click
- tested cancel button redirect on click

### CODE REVIEW

File: frontend/src/components/grantaccess/form/UserNameInput.vue [hunk 1]
```diff
@@ -90,6 +90,7 @@ watch(
                         >{{ helperText }}</small
                     >
                     <ErrorMessage
+                        data-testid="test"
```
Comment (MCatherine1994) [inline frontend/src/components/grantaccess/form/UserNameInput.vue:93]: Hi Jota, just wondering if we need this line? thanks!

File: frontend/src/tests/GrantApplicationAdmin.spec.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    beforeEach(async () => {
+        wrapper = mount(GrantApplicationAdmin, {
+            global: {
+                plugins: [router], // Inject mocked Vue Router
+            },
+        });
+        usernameInputText = wrapper.find('#userIdInput');
+        usernameInputTextEl = usernameInputText.element as HTMLInputElement;
+        verifyButton = wrapper.find("[data-target-btn='verifyIdir']");
+        verifyButtonEl = verifyButton.element as HTMLButtonElement;
```
Comment (MCatherine1994) [inline frontend/src/tests/GrantApplicationAdmin.spec.ts:49]: Hi Jota, I'm not seeing we use the usernameInputText, usernameInputTextEl, verifyButton and verifyButtonEl in multiple test cases, I think we can remove them from here and the above.

File: frontend/src/tests/GrantApplicationAdmin.spec.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+        await cancelBtn.trigger('click');
+
+        // cancelForm is called
+        expect(cancelFormSpy).toHaveBeenCalled();
+        // called route.push with '/dashboard'
+        expect(routerPushSpy).toHaveBeenCalledWith(routeItems.dashboard.path);
+    });
+
+    it('Should submit btn be render and desabled', () => {
```
Comment (MCatherine1994) [inline frontend/src/tests/GrantApplicationAdmin.spec.ts:170]: let's add "disabled by default", thanks!

**Comment by MCatherine1994** [comment]
Hi Jota, looks really good, just small cleanup comment, thanks so much!

File: frontend/src/components/grantaccess/form/UserNameInput.vue (hunk 1 above)
Comment (J0taFerreira) [reply frontend/src/components/grantaccess/form/UserNameInput.vue:93]: Hello, this line should have been cleaned up

File: frontend/src/tests/GrantApplicationAdmin.spec.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    return {
+        ...actual,
+        useRoute: () => {
+            return routes.filter(
+                (route) => route.name == routeItems.grantAppAdmin.name
+            )[0];
+        },
+        useRouter: () => {
+            return {
+                push: mockRoutePush,
```
Comment (MCatherine1994) [inline frontend/src/tests/GrantApplicationAdmin.spec.ts:23]: Hi Jota, sorry just noticed this, do we need this useRouter?

Comment (MCatherine1994) [reply frontend/src/tests/GrantApplicationAdmin.spec.ts:23]: maybe we don't need this and the mockRoutePush, cause I think we have another routerPushSpy beblow

Comment (J0taFerreira) [reply frontend/src/tests/GrantApplicationAdmin.spec.ts:23]: it's not needed anymore, forget to remove after change some code, thanks!

**Review by MCatherine1994 — APPROVED** [review]

**Pull request merged**